ios/RNSScreenContainer.m

```diff
@@ -122,18 +122,51 @@ - (void)updateContainer
   // this is done to mimick the effect UINavigationController has when willMoveToWindow:nil is
   // triggered before the animation starts
   if (activeScreenAdded) {
-    for (RNSScreenView *screen in _reactSubviews) {
-      if (screen.active && [_activeScreens containsObject:screen]) {
-        [self detachScreen:screen];
-        // disable interactions for the duration of transition
-        screen.userInteractionEnabled = NO;
-      }
+    // Following the assumption that there're mostly two active screens
+    // it's reasonable to assume that it's not allowed to add an additional one
+    // if there're already two on them
+    RCTAssert(_activeScreens.count < 2, @"It's allowed to have max two active screens");
+    RNSScreenView* onlyActiveScreen = [_activeScreens anyObject];
+    if (onlyActiveScreen != nil) {
+      // We're not using [self.detachScreen:] couse we don't
+      // want ot actually call [view.removeFromSuperview] in order 
+      // to keep the gesture recognition continuous
+      [onlyActiveScreen.controller removeFromParentViewController];
+      [onlyActiveScreen.controller willMoveToParentViewController:nil];
+      [_activeScreens removeObject:onlyActiveScreen];
+      onlyActiveScreen.userInteractionEnabled = YES;
     }
 
-    // add new screens in order they are placed in subviews array
+    // add new screens in the order they were placed in subviews array
     for (RNSScreenView *screen in _reactSubviews) {
       if (screen.active) {
-        [self attachScreen:screen];
+        if (screen == onlyActiveScreen) {
+          // We're not using [self.attachScreen:] cause we don't
+          // want to call [view.addSubview:] to keep the gesture
+          // recognition continuous
+          [_controller addChildViewController:screen.controller];
+          [screen.controller didMoveToParentViewController:_controller];
+          [_activeScreens addObject:screen];
+        } else {
+          [_controller addChildViewController:screen.controller];
+          if (onlyActiveScreen == nil) {
+            [_controller.view addSubview:screen.controller.view];
+          } else {
+            // We need to keed the native views' order sync with js.
```

Comment:
```suggestion
            // We need to keep the native views' order sync with js.
```

Reply:
Isn't it repeating the same you say later?

Reply:
More or less